### ISSUE TITLE
Assistance with Incorrect TensorRT Inference Results in StreamPetr-trt.

### ISSUE THREAD

**Issue by avi9700** [NVIDIA/DL4AGX#71]
Hi, Thank you for your contribution.
After performing the data dump using the code below, I proceeded with inference and visualization. 
Unfortunately, the results were completely incorrect.
I’m not sure what might be causing this issue. 
Do you have any ideas or suggestions on what I might be doing wrong?
Any guidance would be greatly appreciated!
Thank you,

My GPU environment is as follows.
 - ubuntu 20.04
 - x86
 - cuda : 11.8
 - tensorrt : 8.6.1.6
 - rtx4080

Torch tensors were dumped using the following method.

petr3d.py -> simple_test -> 
        import os
        root = f"dump/{self.count:04d}"
        if not os.path.exists(root):
            os.makedirs(root)        
        data['img].cpu().detach().numpy().tofile(root + '/img.bin')
streampetr_head.py ->forward ->

        import os
        root = f"dump/{self.count:04d}"
        if not os.path.exists(root):
            os.makedirs(root)        
        pos_embed.cpu().detach().numpy().tofile(root + '/pos_embed.bin')
        cone.cpu().detach().numpy().tofile(root + '/cone.bin')
        all_cls_scores.cpu().detach().numpy().tofile(root + '/all_cls_scores.bin')
        all_bbox_preds.cpu().detach().numpy().tofile(root + '/all_bbox_preds.bin')
        timestamp = data['timestamp']
        timestamp.cpu().detach().numpy().tofile(root+'/data_timestamp.bin')

        ego_pose = data['ego_pose']
        ego_pose.cpu().detach().numpy().tofile(root+'/data_ego_pose.bin')

        print("ego_pose = {}".format(ego_pose))

        ego_pose_inv = data['ego_pose_inv']
        ego_pose_inv.cpu().detach().numpy().tofile(root+'/data_ego_pose_inv.bin')

# tools/dist_test.sh projects/configs/StreamPETR/stream_petr_r50_flash_704_bs2_seq_428e_nui_60e.py stream_petr_r50_flash_704_bs2_seq_428e_nui_60e.pth 1 --eval bbox

dump data
https://drive.google.com/file/d/1pkp6mT50KVzY7xANtctZlc1YO8T3WZwv/view?usp=sharing

onnx file
https://drive.google.com/file/d/1AxdDYpM1wxnufBUAFQZDa64xBb9T8n-u/view?usp=sharing

I converted the ONNX file to a TRT engine using the following command.

/home/sml/TensorRT-8.6.1.6/bin/trtexec --onnx=./simplify_extract_img_feat.onnx \
        --fp16 \
        --saveEngine=./simplify_extract_img_feat.engine

/home/sml/TensorRT-8.6.1.6/bin/trtexec --onnx=./simplify_pts_head_memory.onnx \
        --saveEngine=./simplify_pts_head_memory.engine

engine file 
https://drive.google.com/file/d/11W5oTuT-redmzUhsmh8jet7mICssVDn2/view?usp=sharing

inference result :
![Image](https://github.com/user-attachments/assets/c733949f-bf32-403a-9b7c-3e44d36ede8e)

**Comment by zgzhycz** [comment]
I encountered the same issue, could you tell me how to resolve it? thank you!

**Comment by wyd-nb** [comment]
@avi9700 @zgzhycz 
I am currently facing the same problem and would greatly appreciate it if you could share how you resolved it. 
Thank you for your time and assistance!
